services/deployment-agent/Makefile

```diff
@@ -1,98 +1,73 @@
 export VCS_REF:=$(shell git rev-parse --short HEAD)
 export VCS_STATUS_CLIENT:=$(if $(shell git status -s),'modified/untracked','clean')
 export BUILD_DATE:=$(shell date -u +"%Y-%m-%dT%H:%M:%SZ")
 
-# using ?= will only set if absent
-export DOCKER_IMAGE_TAG ?= $(shell cat VERSION)
-$(info DOCKER_IMAGE_TAG set to ${DOCKER_IMAGE_TAG})
-
-# default to local (no registry)
 export DOCKER_REGISTRY ?= itisfoundation
+export DOCKER_IMAGE_TAG ?= $(shell cat VERSION)
 $(info DOCKER_REGISTRY set to ${DOCKER_REGISTRY})
+$(info DOCKER_IMAGE_TAG set to ${DOCKER_IMAGE_TAG})
 
+# STACK_NAME defaults to name of the current directory. Should not to be changed if you follow GitOps operating procedures.
+STACK_NAME = $(notdir $(PWD))
+SWARM_HOSTS = $(shell docker node ls --format={{.Hostname}} 2>/dev/null)
+TEMPCOMPOSE := $(shell mktemp)
 
-## Tools ------------------------------------------------------------------------------------------------------
-#
-tools =
-
-ifeq ($(shell uname -s),Darwin)
-	SED = gsed
-else
-	SED = sed
-endif
-
-ifeq ($(shell which ${SED}),)
-	tools += $(SED)
-endif
-
+.PHONY: help
+help:  ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n\nTargets:\n"} /^[a-zA-Z_-]+:.*?##/ { printf "  \033[36m%-10s\033[0m %s\n", $$1, $$2 }' $(MAKEFILE_LIST)
 
-## ------------------------------------------------------------------------------------------------------
-.PHONY: all
-all: help info
-ifdef tools
-	$(error "Can't find tools:${tools}")
-endif
+.PHONY: info ## target: info – Displays some parameters of makefile environments
+info:
+	@echo '+ VCS_* '
+	@echo '  - URL                : ${VCS_URL}'
+	@echo '  - REF                : ${VCS_REF}'
+	@echo '  - (STATUS)REF_CLIENT : (${VCS_STATUS_CLIENT})'
+	@echo '+ BUILD_DATE           : ${BUILD_DATE}'
+	@echo '+ DOCKER_REGISTRY      : ${DOCKER_REGISTRY}'
+	@echo '+ DOCKER_IMAGE_TAG     : ${DOCKER_IMAGE_TAG}'
 
+.PHONY: init
+init: ## initialize swarm cluster
+	$(if $(SWARM_HOSTS),  \
+		,                 \
+		docker swarm init \
+	)
 
 .PHONY: build build-devel
-# target: build, build-devel: – Builds all service images.
-build:
-	${DOCKER_COMPOSE} -f docker-compose.yml build --parallel
+build: ## target: build, build-devel: – Builds all service images.
+	docker-compose -f docker-compose.yml build --parallel
 
 build-devel:
-	${DOCKER_COMPOSE} -f docker-compose.yml -f docker-compose.devel.yaml build --parallel
+	docker-compose -f docker-compose.yml -f docker-compose.devel.yaml build --parallel
 
-.PHONY: up up-devel down
-# target: up, up-devel, down: – Starts/Stops services.
-up-devel: .env deployment_config.yaml
-	${DOCKER} swarm init
-	${DOCKER_COMPOSE} -f docker-compose.yml -f docker-compose.devel.yaml config > $(TEMPCOMPOSE).tmp-compose.yml
-	${DOCKER} stack deploy -c $(TEMPCOMPOSE).tmp-compose.yml portainer
+.PHONY: up up-devel
+up-devel: init .env deployment_config.yaml ## target: up, up-devel, down: – Starts/Stops services.
```

Comment:
Minor: no need to write "target" in all the help entries to get it identified as help. Also, every target can has its own input

Reply:
done